bleak/backends/corebluetooth/client.py

```diff
@@ -80,12 +79,17 @@ async def connect(self, **kwargs) -> bool:
                 raise BleakError(
                     "Device with address {} was not found".format(self.address)
                 )
-
-        logger.debug("Connecting to BLE device @ {}".format(self.address))
-
-        manager = self._device_info.manager().delegate()
-        await manager.connect_(self._device_info)
-        manager.disconnected_callback = self._disconnected_callback_client
+        # self._device_info.manager() should return a CBCentralManager
```

Comment:
I think a simpler change would be:

```diff
diff --git a/bleak/backends/corebluetooth/client.py b/bleak/backends/corebluetooth/client.py
index 954479d..093695b 100644
--- a/bleak/backends/corebluetooth/client.py
+++ b/bleak/backends/corebluetooth/client.py
@@ -73,7 +73,7 @@ class BleakClientCoreBluetooth(BaseBleakClient):
                 self.address, timeout=timeout
             )

-            if device:
+            if device and device.details:
                 self._device_info = device.details
             else:
                 raise BleakError(
```

But the deeper question is why `device.details` would be `None`.